test/formats.js

```diff
@@ -0,0 +1,480 @@
+const SSBURI = require('ssb-uri2')
+const bendyButt = require('ssb-bendy-butt')
+const { where, author, toPromise } = require('ssb-db2/operators')
+
+function createSsbServer() {
+  return SecretStack({ appKey: caps.shs })
+    .use(require('ssb-db2'))
+    .use(require('ssb-db2/compat/ebt'))
+    .use(require('../'))
+}
+
+const CONNECTION_TIMEOUT = 500 // ms
+const REPLICATION_TIMEOUT = 2 * CONNECTION_TIMEOUT
+
+const aliceDir = '/tmp/test-format-alice'
+rimraf.sync(aliceDir)
+mkdirp.sync(aliceDir)
+
+let alice = createSsbServer().call(null, {
+  path: aliceDir,
+  timeout: CONNECTION_TIMEOUT,
+  keys: u.keysFor('alice'),
+})
+
+const bobDir = '/tmp/test-format-bob'
+rimraf.sync(bobDir)
+mkdirp.sync(bobDir)
+
+let bob = createSsbServer().call(null, {
+  path: bobDir,
+  timeout: CONNECTION_TIMEOUT,
+  keys: u.keysFor('bob')
+})
+
+const bendyButtMethods = {
+  // used in request, block, cleanClock, sbot.post
+  isFeed: SSBURI.isBendyButtV1FeedSSBURI,
+  getAtSequence(sbot, pair, cb) {
+    sbot.getAtSequence([pair.id, pair.sequence], (err, msg) => {
+      cb(err, msg ? bendyButt.encode(msg.value) : null)
+    })
+  },
+  appendMsg(sbot, msgVal, cb) {
+    sbot.add(bendyButt.decode(msgVal), (err, msg) => {
+      cb(err && err.fatal ? err : null, msg)
+    })
+  },
+
+  // used in ebt:stream to distinguish between messages and notes
+  isMsg(bbVal) {
+    if (!Buffer.isBuffer(bbVal)) return false
+
+    const msgVal = bendyButt.decode(bbVal)
+    return msgVal &&
+      Number.isInteger(msgVal.sequence) && msgVal.sequence > 0 &&
+      typeof msgVal.author == 'string' && msgVal.content
+  },
+  // used in ebt:events
+  getMsgAuthor(bbVal) {
+    //console.log("bb getMsgAuthor", Buffer.isBuffer(bbVal))
+    if (Buffer.isBuffer(bbVal))
+      return bendyButt.decode(bbVal).author
+    else
+      return bbVal.author
+  },
+  // used in ebt:events
+  getMsgSequence(bbVal) {
+    //console.log("bb getMsgSequence", Buffer.isBuffer(bbVal))
+    if (Buffer.isBuffer(bbVal))
+      return bendyButt.decode(bbVal).sequence
+    else
+      return bbVal.sequence
+  }
+}
+
+function getBBMsg(mainKeys) {
+  // fake some keys
+  const mfKeys = ssbKeys.generate()
+  const classicUri = SSBURI.fromFeedSigil(mfKeys.id)
+  const { type, /* format, */ data } = SSBURI.decompose(classicUri)
+  const bendybuttUri = SSBURI.compose({ type, format: 'bendybutt-v1', data })
+  mfKeys.id = bendybuttUri
+
+  const content = {
+    type: "metafeed/add",
+    feedpurpose: "main",
+    subfeed: mainKeys.id,
+    metafeed: mfKeys.id,
+    tangles: {
+      metafeed: {
+        root: null,
+        previous: null
+      }
+    }
+  }
+
+  const bbmsg = bendyButt.encodeNew(
+    content,
+    mainKeys,
+    mfKeys,
+    1,
+    null,
+    Date.now(),
+    null
+  )
+
+  return bendyButt.decode(bbmsg)
+}
+
+// need them later
+let aliceMFId
+let bobMFId
+
+tape('multiple formats', async (t) => {
+  alice.ebt.registerFormat('bendybutt', bendyButtMethods)
```

Comment:
Can we use the constants from ssb-uri here?

so `bendybutt-v1` instead of this and `ed25519` instead of `classic`?

Reply:
Good idea, thanks for the input :)

Reply:
@arj03 If it helps, I can start renaming `classic` to `ed25519` in ssb-meta-feeds and downstream

Reply:
Aaactually, ssb-bfe-spec relies on `classic` https://github.com/ssb-ngi-pointer/ssb-bfe-spec/blob/a4e38c30b67d7110115c7fc8440b71448f87d8d2/bfe.json#L6 so we could keep it `classic` while rename `bendybutt` to `bendybutt-v1`.

Reply:
Personally I prefer the `classic` name :)